service/matching/db_task_queue_ownership.go

```diff
@@ -59,6 +65,10 @@ type (
 		getLastAllocatedTaskID() int64
 		persistTaskQueue(ctx context.Context) error
 		flushTasks(ctx context.Context, taskInfos ...*persistencespb.TaskInfo) error
+		GetVersioningData(ctx context.Context) (*persistencespb.VersioningData, error)
+		MutateVersioningData(ctx context.Context, mutator func(*persistencespb.VersioningData) error) (*persistencespb.VersioningData, error)
+		setVersioningDataForNonRootPartition(verDat *persistencespb.VersioningData)
```

Comment:
what are these?
if the is for task queue versioning, then i would expect the version being part of task queue key, so we handle the task queue partition & task queue version the same way

Reply:
Here I'm just migrating what David & Spencer did in db.go to this file. This function is used for setting the versioning metadata read from root partition for non root partition.

Reply:
@wxing1292 The data is only ever stored on the root partition to avoid duplication